environment/stats/src/cloud/benchflow/collectors/stats.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+	"sync"
+	"github.com/fsouza/go-dockerclient"
+	"github.com/minio/minio-go"
+)
+
+type Container struct {
+	ID           string
+	statsChannel chan *docker.Stats
+	doneChannel chan bool
```

Comment:
Already tried that, still gives me the same error, as if I was closing a channel that's already closed. I'll try testing this separately and see if the issue is somewhere in my code structure, but if it isn't then I'll open an issue on the docker client API github and see if they have an answer, and if it's a bug of their API they can fix it.